sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -602,6 +602,27 @@ pi_result OCL(piKernelCreate)(pi_program program, const char *kernel_name,
   return ret_err;
 }
 
+pi_result OCL(piKernelGetSubGroupInfo)(
+    pi_kernel kernel, pi_device device, pi_kernel_sub_group_info param_name,
+    size_t input_value_size, const void *input_value, size_t param_value_size,
+    void *param_value, size_t *param_value_size_ret) {
+
+  size_t ret_val;
+  cl_int ret_err;
+  ret_err = cast<pi_result>(clGetKernelSubGroupInfo(
+      cast<cl_kernel>(kernel), cast<cl_device_id>(device),
+      cast<cl_kernel_sub_group_info>(param_name), input_value_size, input_value,
+      sizeof(size_t), &ret_val, param_value_size_ret));
+
+  if (ret_err != CL_SUCCESS)
+    return cast<pi_result>(ret_err);
+
+  *((uint32_t *)param_value) = (uint32_t)ret_val;
```

Comment:
the OpenCL returns "size_t", why are you truncating this to 32-bit? If really needed (I don't think it is) we should document this for piKernelGetSubGroupInfo

Reply:
The return type of the get_sub_group_info APIs have been changed to return uint32_t data instead of size_t, which is why this change was done. How should this be documented?

Reply:
Add a comment in the pi.h for piKernelGetSubGroupInfo to indicated now supported parameters with their sizes.

Reply:
But I suggest that we continue to use "size_t" in PI, and truncate at SYCL level.

Reply:
Why is that?

Reply:
I do not mind documenting the change. 
Is that a common practice to make PI API behave different from what is required in SYCL spec? We have kept PI APIs consistent with SYCL Spec, and made any required changes in Plugin.

Reply:
@garimagu 
1) do you know why `size_t` was changed to `uint32_t` in the first place?
2) Can we do C++-style casts here?

Reply:
1. I had asked about this change. The reason given to me was that size_t can change between 64 bits and 32 bits, they wanted the return to be fixed.
" One issue with size_t is it was meant for sizing pointers and for 32-bit builds size_t is 4 bytes instead of 8. I don't know that we'll have 32-bit builds but we would like to be explicit for the size we need for fields in our structures and that it will stay fixed."
2. Yes, I can use C++ style casts.

Reply:
I have made the call to PI_API check the returned size of the result value and truncate the value appropriately. Does this look better?
I'm essentially hesitant to make it size_t based on OpenCL spec, because if any new backend is developed based on the SYCL extension spec, it will have to change the Plugin to cast the uint32_t to size_t first, and then in the library will be casting the size_t to uint32_t. It will be weird that the PI APIs are not SYCL extension spec compliant.

Reply:
> I'm essentially hesitant to make it size_t based on OpenCL spec

I'd be fine making it `uint32_t` in PI, document this in the pi.h, and truncate in OpenCL plugin.
But we should stick to just one size that fits SYCL well.

Reply:
@smaslov-intel : Do you mean the initial solution was ok with the document change?

Reply:
Yes, I'd be fine with it.

Reply:
Thanks. I have updated the review. Let me know if it looks good.